### PR TITLE
Update bobtimus api and extract helper in waves

### LINKED ISSUE #77: Add api prefix to bobtimus endpoints

### PR DESCRIPTION
**Pull request by bonomat**
This PR adds `api` prefix to all Bobtimus API. I've extracted a helper file `Bobtimus.tsx` in waves which should serve as a onestop place to interact with Bobtimus.

It also enables all cors headers for `fake_bobtimus` which simplifies development. 

Resolves #77

### CODE REVIEW

File: waves/src/Bobtimus.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+export async function postSellPayload(payload: CreateSwapPayload) {
+    let res = await fetch("/api/swap/lbtc-lusdt/sell", {
+        method: "POST",
+        headers: {
+            "Content-Type": "application/json",
+            Accept: "application/json",
+        },
+        body: JSON.stringify(payload),
+    });
+    console.log(res);
```
Comment (thomaseizinger) [inline waves/src/Bobtimus.tsx:20]: This looks like a left-over!

File: waves/src/Bobtimus.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import { CreateSwapPayload } from "./wasmProxy";
+
+export async function fundAddress(address: string): Promise<any> {
+    await fetch("/api/faucet/" + address, {
+        method: "POST",
+    });
+}
+
+export async function postSellPayload(payload: CreateSwapPayload) {
+    let res = await fetch("/api/swap/lbtc-lusdt/sell", {
```
Comment (thomaseizinger) [inline waves/src/Bobtimus.tsx:12]: This route doesn't match with what bobtimus listens on? It also didn't match before but I just saw it now!

File: waves/src/index.tsx [hunk 3]
```diff
@@ -1,18 +1,16 @@
...
 import reportWebVitals from "./reportWebVitals";
 import theme from "./theme";
 
 ReactDOM.render(
     <React.StrictMode>
         <ChakraProvider theme={theme}>
-            <SSEProvider endpoint="/rate/lbtc-lusdt">
-                <App />
-            </SSEProvider>
+            <BobtimusRateProvider children={<App />} />
```
Comment (thomaseizinger) [inline waves/src/index.tsx:13]: `children` are usually not passed as literal props. You can do this instead:

```tsx
<BobtimusRateProvider>
	<App />
</BobtimusRateProvider>
```

File: bobtimus/src/bin/fake_bobtimus.rs [hunk 4]
```diff
@@ -43,7 +43,7 @@ async fn main() -> Result<()> {
     let cors = warp::cors().allow_any_origin();
 
     let faucet = warp::post()
-        .and(warp::path("faucet"))
+        .and(warp::path!("api" / "faucet" / ..))
```
Comment (thomaseizinger) [inline bobtimus/src/bin/fake_bobtimus.rs:46]: I find it confusing to mix the macro-based filter (`path!`) with normal ones and it seems to require you to do this `..` stuff.

If you want to use the macro, you [should](https://docs.rs/warp/0.2.5/warp/macro.path.html#example) be able to do this:

```rust
warp::path!("api" / "faucet" / Address)
```

Alternatively, you should be able to use two, chained `path()` functions.

**Review by thomaseizinger — APPROVED** [review]

File: waves/src/Bobtimus.tsx (hunk 2 above)
Comment (bonomat) [reply waves/src/Bobtimus.tsx:12]: Maybe I'm missing something, but I think it is correct: 
https://github.com/comit-network/droplet/blob/3c1b8e3cd804d45981b7fecd9d41615e516a1235/bobtimus/src/http.rs#L35-L40

File: bobtimus/src/bin/fake_bobtimus.rs (hunk 4 above)
Comment (bonomat) [reply bobtimus/src/bin/fake_bobtimus.rs]: cool, works.

File: waves/src/index.tsx (hunk 3 above)
Comment (bonomat) [reply waves/src/index.tsx:13]: Ah, I think now I got it working .

File: waves/src/Bobtimus.tsx (hunk 2 above)
Comment (thomaseizinger) [reply waves/src/Bobtimus.tsx:12]: Ah my bad, I was looking at the `rate` endpoint :)

**Review by thomaseizinger — APPROVED** [review]

**Pull request merged**